autopush/endpoint.py

```diff
@@ -873,11 +902,15 @@ def _success(self, result):
     #                    Utility Methods
     #############################################################
     def _validate_auth(self, uaid):
-        """Validates the Authorization header in a request"""
+        """Validates the Authorization header in a request
+
+        Validate the given request using HAWK.
+        """
         secret = self.ap_settings.crypto_key
-        hashed = self.request.headers.get("Authorization", "").strip()
-        key = generate_hash(secret, uaid)
-        return validate_hash(key, self.request.body, hashed)
+
+        return hawkauthlib.check_signature(self.request,
```

Comment:
It looks like `check_signature` passes the request to `utils.normalize_request_object`. I wonder if we need to pass a facade instead.